teos/src/gatekeeper.rs

```diff
@@ -175,7 +175,7 @@ impl Gatekeeper {
                     .available_slots
                     .checked_add(self.subscription_slots)
                     .ok_or(MaxSlotsReached)?;
-                user_info.subscription_expiry = block_count + self.subscription_duration;
+                user_info.subscription_expiry += self.subscription_duration;
```

Comment:
You might want to do a `checked_add` like the `available_slots` above.

Reply:
Added `checked_add` with a default to `u32::MAX` if there's an overflow.